cmd/thanos/query.go

```diff
@@ -219,6 +223,25 @@ func registerQuery(app *extkingpin.App) {
 			fileSD = file.NewDiscovery(conf, logger)
 		}
 
+		endpointConfigYAML, err := endpointConfig.Content()
+		if err != nil {
+			return err
+		}
+
+		if *secure && len(endpointConfigYAML) != 0 {
+			return errors.Errorf("deprecated flags --grpc-client-tls* and new --endpoint.config flag cannot be specified at the same time; use either of those")
+		}
+
+		var fileSDConfig httpconfig.FileSDConfig
```

Comment:
So this is not being used and the code to process the config was commented out. This would basically disabled the global sd file functionality, no? Also, why use httpconfig.FileSDConfig instead of file.SDConfig?🙂